### PR TITLE
nonlinsolve returns duplicate solution

### PR DESCRIPTION
**Pull request by OmPrakashSingh1704**
cf #13961 

* solvers
     * Added 'expand_and_simplify()' for ignoring repeated solutions in solveset

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
> The following commits **add new files**:

The commits will need to be squashed to remove this.

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -3885,6 +3885,8 @@ def _handle_poly(polys, symbols):
 
         if inexact:
             poly_eqs = [nfloat(p) for p in poly_eqs]
+    # poly_sol=expand(x) for x in poly_sol
```
Comment (samithkavishke) [inline sympy/solvers/solveset.py:3888]: remove these comments

**Review by samithkavishke — CHANGES_REQUESTED** [review]

**Comment by HAWK1704** [comment]
I want to ask that should I write a separate function for expand the solution terms using recursion or should I leave the solution here?

If you want me to write a function then can you provide me some data like how we are expecting the solutions, like the solution I worked on was FiniteSet(Tuple()) type. What I mean is that can I expect solutions like FiniteSet(Tuple(Tuple())), FiniteSet(Tuple(Tuple(Tuple()))), etc,.

**Comment by HAWK1704** [comment]
> I want to ask that should I write a separate function for expand the solution terms using recursion or should I leave the solution here?
> 
> If you want me to write a function then can you provide me some data like how we are expecting the solutions, like the solution I worked on was FiniteSet(Tuple()) type. What I mean is that can I expect solutions like FiniteSet(Tuple(Tuple())), FiniteSet(Tuple(Tuple(Tuple()))), etc,.

Worked on this Idea so verify it once.

**Comment by smichr** [comment]
There should be a test for which the solution would have duplicates in master but not in this branch. Fictitious example:
```python
def test_no_dup-nonlinsolve():
    assert nonlinsolve(x**2-4) == {-2, 2}  <-- in master {-2, 2, 2*(cos(1)**2 + sin(1)**2)}
```

**Review by samithkavishke — CHANGES_REQUESTED** [review]
You must add a test case to verify this code.

**Comment by HAWK1704** [comment]
I'll look into that after my exams over.

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -3888,6 +3888,21 @@ def _handle_poly(polys, symbols):
 
     return poly_sol, poly_eqs
 
+def expand_and_simplify(subs_res):
```
Comment (smichr) [inline sympy/solvers/solveset.py:3891]: `simplify` should not be used (in general) to fix very specific problems. As pointed out in the issue, the problem stems from how 2-arg Adds are handled. Apparently, two different code paths produced the same solution but did so in a different way and one leaves a negative one factored out while the other has it distributed. One solution is to just patch things up here with a routine like
```python
def touchup(sol):
    # put the minimal simplification needed to keep duplicate
    # solutions from appearing in the solution
    return bottom_up(sol, signsimp)
```
(and import bottom_up on the line importing from traversal).

A better solution would be to track down why these two expressions are being constructed in two different ways and see if they should be doing things the same way or using common code.

The best solution would be to disable distribution of the constant onto an Add (but that is a very involved task).

**Comment by HAWK1704** [comment]
how to resolve ``doctest-latest`` and ``authors`` failure?

**Pull request closed without merging**